### PR TITLE
Check for missing i18n strings in CI

### PR DESCRIPTION
**Pull request by nschonni**
Had a separate PR to do a large sync with the `add-missing`. This just checks the `en` locale and fails if that sync creates a diff.
Pushing the CI as first commit, then will add the en-only fixup in a following commit to confirm it goes red->green

### CODE REVIEW

**Comment by nschonni** [comment]
If this makes sense, the `-l en` could be removed from these tasks after #23367

**Review by ykzts — APPROVED** [review]

File: config/locales/en.yml [hunk 1]
```diff
@@ -473,6 +476,7 @@ en:
       content_policies:
         comment: Internal note
         description_html: You can define content policies that will be applied to all accounts from this domain and any of its subdomains.
+        limited_federation_mode_description_html: Limited federation mode description html
```
Comment (Gargron) [inline config/locales/en.yml:479]: There gotta be some real text here but i don’t know what it is.

**Review by Gargron — CHANGES_REQUESTED** [review]

File: config/locales/en.yml (hunk 1 above)
Comment (nschonni) [reply config/locales/en.yml:479]: This was the only reference I saw doing a quick code search https://github.com/mastodon/mastodon/blob/6a5e447753095606e42cc3899887ba7fae77a0e8/app/views/admin/instances/show.html.haml#L40
I think there may be a better search with the `i18n-tasks` cli though

Comment (nschonni) [reply config/locales/en.yml:479]: Maybe that reference should be to `admin.instances.content_policies.description_html` instead

Comment (ClearlyClaire) [reply config/locales/en.yml:479]: I don't think so, the missing string is specifically for the limited federation mode. The reference was added in https://github.com/mastodon/mastodon/pull/17704 but without the actual string.

I think it should say something like “You are currently allowing federation with this domain.”

**Review by KittyKrusad3r — APPROVED** [review]

**Review by KittyKrusad3r — APPROVED** [review]

File: config/locales/en.yml (hunk 1 above)
Comment (nschonni) [reply config/locales/en.yml:479]: Do you want to add that as a suggestion for me to commit, or does it need a little more digging?

Comment (ClearlyClaire) [reply config/locales/en.yml:479]: I mean the string didn't exist before #17704 and it was either never written at that point, or left out of the commit, so Gargron's original intent is only known to him. Looking at the code a bit more the string *might* be shown when a domain is not allowed, so my previous suggestion may not be correct. I'd suggest the following instead:

```suggestion
        limited_federation_mode_description_html: You are chose whether to allow federation with this domain.
```

File: config/locales/en.yml [hunk 2]
```diff
@@ -473,6 +476,7 @@ en:
       content_policies:
         comment: Internal note
         description_html: You can define content policies that will be applied to all accounts from this domain and any of its subdomains.
+        limited_federation_mode_description_html: You are chose whether to allow federation with this domain.
```
Comment (ClearlyClaire) [inline config/locales/en.yml:479]: I'm sorry I rewrote this mid-sentence, I didn't notice I ended up with a broken sentence.

```suggestion
        limited_federation_mode_description_html: You can chose whether to allow federation with this domain.
```

**Review by ClearlyClaire — CHANGES_REQUESTED** [review]

**Review by ClearlyClaire — APPROVED** [review]

**Comment by nschonni** [comment]
@ykzts I notice you run something to normalize the JSON files on the Crowdin PRs. Is that something that should also be integrated here?

**Comment by ClearlyClaire** [comment]
@Gargron I can't merge it because you have requested changes and not reviewed/accepted them.

**Comment by nschonni** [comment]
The CircleCI failure seems unrelated. I can't seem to restart that job, but I can rebae again if needed to trigger it again

**Review by Gargron — APPROVED** [review]

**Pull request merged**